File: src/models/FrequentlyAskedQuestion.ts
class FrequentlyAskedQuestion{
    question:String|null;
    answer:String|null;
    created_at:Date|null;
    updated_at:Date|null;
    deleted_at:Date|null;
    menu_id:number|null;
    id:number|null;

    constructor(
        question:String|null,
        answer:String|null,
        created_at:Date|null, 
        updated_at:Date|null, 
        deleted_at:Date|null,
        menu_id:number|null=null,
        id:number|null=null,
    ){
        this.question = question;
        this.answer = answer;
        this.created_at = created_at;
        this.updated_at = updated_at;
        this.deleted_at = deleted_at;
        this.menu_id = menu_id;
        this.id = id;
    }
};
export default FrequentlyAskedQuestion;

File: src/main/java/pe/tarjetaestilos/models/FrequentlyAskedQuestion.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package pe.tarjetaestilos.models;

/**
 *
 * @author julito
 */
import com.fasterxml.jackson.annotation.JsonBackReference;
import java.time.ZonedDateTime;
import java.util.List;
import java.util.ArrayList;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import org.hibernate.annotations.Where;

@Entity
@Table(name = "frequently_asked_questions")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Where(clause = "deleted_at IS NULL")
public class FrequentlyAskedQuestion {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String question;
    private String answer;
    
    @ManyToOne()
    @JoinColumn(name = "menu_id", nullable = false)
    @ToString.Exclude
    @JsonBackReference
    private Menu menu;
    
    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private ZonedDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private ZonedDateTime updatedAt;

    @Column(name = "deleted_at")
    private ZonedDateTime deletedAt;
}
